app/src/main/java/ch/hikemate/app/model/profile/Profile.kt

```diff
@@ -1,12 +1,21 @@
 package ch.hikemate.app.model.profile
 
+import ch.hikemate.app.R
 import com.google.firebase.Timestamp
 
 /** An enum class representing the hiking level of a user. */
 enum class HikingLevel {
   BEGINNER,
   INTERMEDIATE,
-  EXPERT
+  EXPERT;
+
+  fun getDisplayNameId(): Int {
+    return when (this) {
+      BEGINNER -> R.string.hiking_level_beginner
+      INTERMEDIATE -> R.string.hiking_level_intermediate
+      EXPERT -> R.string.hiking_level_expert
+    }
```

Comment:
Suggestion: Nice, this is a way better placement for this function. 

However, I don't think the way the function currently needs to be used (first fetching id and then fetching the string that belongs to this id) is very intuitive. I've added a suggestion, which takes the context and returns a string directly, thus avoiding the detour with the StringID. If you accept the suggestion, make sure to also update the usages of this function in the code.

I'll leave it up to you to see which option you prefer, since it works the way it currently is too, but either way, please make sure to add a docstring to document this functions use, so that we can keep our code nice and understandable and maintainable.

```suggestion
  fun getDisplayString(context: Context): String {
    return when (this) {
      BEGINNER -> context.getString(R.string.hiking_level_beginner)
      INTERMEDIATE -> context.getString(R.string.hiking_level_intermediate)
      EXPERT -> context.getString(R.string.hiking_level_expert)
    }
```

Reply:
Yes, I agree with you. I changed it

Reply:
Thanks. Could you please add a docstring as well? I am sorry for being so nitpicky, but thats a feedback we got from the coaches.

Reply:
Sure no worries!